### PR TITLE
Do not send sampleRate when it is 100 #74

### PR DESCRIPTION
**Pull request by kamilszostak**
* Refactor aiDataContract object to use enum instead of bool 
* User can use functions in aiDataContract (see Envelope.ts)
* User can mark a fields as hidden an it will not be serialized

### CODE REVIEW

File: JavaScript/JavaScriptSDK/Telemetry/Common/Envelope.ts [hunk 1]
```diff
@@ -20,14 +20,16 @@ module Microsoft.ApplicationInsights.Telemetry.Common {
...
-                iKey: true,
-                name: true,
-                sampleRate: true,
-                tags: true,
-                data: true
+                time: FieldType.Required,
+                iKey: FieldType.Required,
+                name: FieldType.Required,
+                sampleRate: () => {
+                    return (this.sampleRate == 100) ? FieldType.Hidden : FieldType.Required;
```
Comment (SergeyKanzhelev) [inline JavaScript/JavaScriptSDK/Telemetry/Common/Envelope.ts]: maybe you need to compare with precision. Not sure how JavaScript handles doubles

**Comment by MaxShehovtsov** [comment]
:shipit:

**Comment by SergeyKanzhelev** [comment]
@MaxShehovtsov so in javascript comparing double (or float) with integer will work just fine? Do we need to compare with precision?

**Comment by MaxShehovtsov** [comment]
@SergeyKanzhelev there's only "number" type in javascript (http://www.ecma-international.org/ecma-262/5.1/#sec-4.3.20) . See this example: http://jsfiddle.net/zoyt3tpn/2/

**Comment by kamilszostak** [comment]
It looks like JS handles float numbers quite well and we don't need to put any special code.

See: http://jsfiddle.net/zoyt3tpn/11/

**Comment by MaxShehovtsov** [comment]
@kamilszostak your example actually disproves it (given the 2nd 'false') :) But I still think it's fine :) 
:sheep: :it:

**Pull request merged**